### PR TITLE
Revamp `basic.md` (put it under the `getting_started` section)

### PR DESCRIPTION
**Pull request by Saransh-cpp**
This PR revamps `basic.md` and renames it to "Neural Networks from scratch". Most of the content is intact, but I have cleaned up the code and removed the bits and pieces which are not supposed to be used now.

Note: This PR contains the changes from the linear regression PR and the logistic regression PR (without some of these changes, the tests would fail). This PR should be merged after both of these PRs. I will make sure to sync the changes before this PR is merged.

Lastly, I am not sure if this guide should contain a training section. The title talks only about constructing Neural Networks, but I can add a training function and showcase `Optimisers.jl`.

Rendered guide - https://github.com/Saransh-cpp/Flux.jl/blob/revamp-basic/docs/src/getting_started/neural_nets_from_scratch.md

### PR Checklist

- [ ] Tests are added
- [ ] Entry in NEWS.md
- [ ] Documentation, if applicable

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
The docs now have a new and revamped `basic` section named "[How Flux Works: Gradients and Layers](https://fluxml.ai/Flux.jl/stable/models/basics/#man-basics)".

Closing this PR as the current section looks great!
